engine/pom.xml

```diff
@@ -99,6 +99,18 @@
             <artifactId>money-api</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <!-- We exclude this dependency from the classpath when running tests with OracleJDK 8,
+                 because OracleJDK 8 already bundles JavaFX. -->
+            <groupId>org.openjfx</groupId>
+            <artifactId>javafx-base</artifactId>
+            <!-- The JavaFX dependency is optional (not pulled transitively by our consumers) for two reasons:
+                 1. HV should work fine without it, as long as you don't use JavaFX.
+                 2. We don't want to force the dependency on our users who are still on OracleJDK 8,
+                    since OracleJDK 8 already bundles JavaFX.
+              -->
+            <optional>true</optional>
+        </dependency>
```

Comment:
Be aware of this particular change.

I don't think it changes much for consumers of HV: the dependency is marked as optional, which means it won't transitively "contaminate" consuming projects.

However, the dependency is now necessary when testing with JDK8 as well as when testing with JDK11, because we never compile with JDK8 (we just run the tests with JDK8). So it's just simpler to always include the dependency, and add classpath exclusions to remove it at runtime when testing with JDK8 (see the maven-surefire-plugin configuration changes in the root POM).

If this optional dependency is a problem, I suppose we could just add it to both profiles (`testWithJdk8` and `testWithJdk11+`). Then consuming projects would not be affected at all then. But... this seems unnecessary.